azurerm/internal/services/eventgrid/eventgrid_system_topic_resource.go

```diff
@@ -0,0 +1,219 @@
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.EventGridSystemTopicID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"source": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"topic_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Microsoft.Eventhub.Namespaces",
+					"Microsoft.Storage.StorageAccounts",
+					"Microsoft.Resources.Subscriptions",
+					"Microsoft.Resources.ResourceGroups",
+					"Microsoft.Devices.IoTHubs",
+					"Microsoft.EventGrid.Topics",
+					"Microsoft.ServiceBus.Namespaces",
+					"Microsoft.ContainerRegistry.Registries",
+					"Microsoft.Media.MediaServices",
+					"Microsoft.Maps.Accounts",
+					"Microsoft.EventGrid.Domains",
+					"Microsoft.AppConfiguration.ConfigurationStores",
+					"Microsoft.KeyVault.vaults",
+					"Microsoft.Web.Sites",
+					"Microsoft.Web.ServerFarms",
+					"Microsoft.SignalRService.SignalR",
+					"Microsoft.MachineLearningServices.Workspaces",
+					"Microsoft.Communication.CommunicationServices",
+				}, false),
+			},
+
+			"metric_resource_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmEventGridSystemTopicCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).EventGrid.SystemTopicsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	source := d.Get("source").(string)
+	topicType := d.Get("topic_type").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing EventGrid System Topic %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_eventgrid_system_topic", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	t := d.Get("tags").(map[string]interface{})
+
+	systemtopicProperties := &eventgrid.SystemTopicProperties{
+		Source:    &source,
+		TopicType: &topicType,
+	}
+
+	systemTopic := eventgrid.SystemTopic{
+		Location:              &location,
+		SystemTopicProperties: systemtopicProperties,
+		Tags:                  tags.Expand(t),
+	}
+
```

Comment:
couldw e combine these
```suggestion

	systemTopic := eventgrid.SystemTopic{
		Location:              &location,
		SystemTopicProperties: &eventgrid.SystemTopicProperties{
			Source:    &source,
			TopicType: &topicType,
		},
		Tags:                  tags.Expand(t),
	}

```